### ISSUE TITLE
TApplication.HandleException Access violation

### ISSUE THREAD

**Issue by traumschule** [tomboy-notes/tomboy-ng#2]
Just hit the first crash with tomboy-ng in the background, without really working with it. Will see if it is reproducible.

About output: Build date 2017/11/26  TargetCPU i386  OS Linux
OS: debian stretch
Version: ea481bdbfa133b7a3de75b63042440674b149da3
size: 32M
build command: `lazbuild --pcp=../../lazconf Tomboy_NG.lpi # lazconf contains environmentoptions.xml (automatically copied from /etc/lazarus/) and packagefiles.xml which links kcontrolsbase.lpk kcontrolslaz.lpk`
Lazarus version: 1.8.0RC5

> Peek !!!
> Peek !!!
> Note has bad xml/home/user/src/tomboy/Notes/3be61374-8545-4852-b5cc-11971e689c7f.note
> Note has no Title
> Note has bad xml/home/user/src/tomboy/Notes/3be61374-8545-4852-b5cc-11971e689c7f.note
> Note has no Title
> Note has bad xml/home/user/src/tomboy/Notes/3be61374-8545-4852-b5cc-11971e689c7f.note
> Note has bad xml/home/user/src/tomboy/Notes/3be61374-8545-4852-b5cc-11971e689c7f.note
> Peek !!!
> Peek !!!
> TApplication.HandleException Access violation
>   Stack trace:
>   $08317DE6  TNOTELISTER__THISNOTEISOPEN,  line 339 of note_lister.pas
>   $08097D13  TRTSEARCH__NOTECLOSING,  line 171 of mainunit.pas
>   $08319334  TEDITBOXFORM__FORMDESTROY,  line 581 of editbox.pas
>   $08087867  TCUSTOMFORM__DODESTROY,  line 976 of ./include/customform.inc
>   $08085EC6  TCUSTOMFORM__BEFOREDESTRUCTION,  line 184 of ./include/customform.inc
>   $08085EF0  TCUSTOMFORM__DESTROY,  line 198 of ./include/customform.inc
>   $08070A32
>   $0808223D  BEFOREFINALIZATION,  line 1904 of forms.pp
>   $0807402D
> Exception at 08317DE6: EAccessViolation:
> Access violation.
> Heap dump by heaptrc unit
> 145072556 memory blocks allocated : 2119444449/2896017840
> 145072556 memory blocks freed     : 2119444449/2896017840
> 0 unfreed memory blocks : 0
> True heap size : 688128
> True free heap : 688128

edit: oh and i found a note is missing, maybe it crashed trying to save it.

**Comment by davidbannon** [comment]
This looks, to me, as if "something" happened, that particular note decided it needed to be saved and we ran into the same problem that caused the exception in #14. Thats won't happen again but what was the "something" that caused the shutdown in the first place ?

Its particularly  disturbing in that it lost a note. I was pretty sure thats not likely to happen anymore. One thing, I thought at one time I saw a note being saved on exit that had not been changed. Did not chase that up at the time, it could be related but is not the "something" that triggered off the problem. Interesting that you were not interacting with tomboy-ng at the time, once changed notes are saved, it really should be doing nothing .....

If you see it happen again, be interesting to dee what dmsg says, OOM killer is a (remote) possibility.
